src/bootstrap/src/core/build_steps/doc.rs

```diff
@@ -830,7 +830,8 @@ impl Step for Rustc {
         cargo.rustdocflag("--show-type-layout");
         // FIXME: `--generate-link-to-definition` tries to resolve cfged out code
         // see https://github.com/rust-lang/rust/pull/122066#issuecomment-1983049222
```

Comment:
Is there an issue tracking getting rid of the cfged-out-code doc generating?

Reply:
Not that I know of.

""Normally"" we're supposed to do that already. The HIR visitor might provide it though (which is why it failed in the past), would be interesting to check why.

Reply:
So it might be that the cfg doc is not necessary? Can you test that? If it is necessary, please create a tracking issue

Reply:
The cfg doc is necessary, otherwise I get:

```
error[E0433]: failed to resolve: could not resolve path `rustc_passes::hir_id_validator::check_crate`
   --> compiler/rustc_interface/src/passes.rs:833:5
    |
833 |     rustc_passes::hir_id_validator::check_crate(tcx);
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ could not resolve path `rustc_passes::hir_id_validator::check_crate`
    |
```

Opened https://github.com/rust-lang/rust/issues/136604.